liiatools_pipeline/ops/school_census_org.py

```diff
@@ -0,0 +1,332 @@
+
+    # Create empty dict to store outputs
+    outputs = {}
+
+    # On roll tables
+    # Make demographics output
+    pupil_on = open_file(session_folder, "school_census_pupilonroll.csv")
+
+    # Open addresses files, where there may be one or two files to open
+    addresses_pattern = re.compile(r"school_census_addresses\.csv$")
+    addresses_onroll_pattern = re.compile(r"school_census_addressesonroll\.csv$")
+    addresses_patterns = [addresses_pattern, addresses_onroll_pattern]
+    addresses_on = open_and_concat_patterns(files, addresses_patterns, session_folder)
+    addresses_on = addresses_on.drop(columns=["addressesorderseqcolumn", "addressesonrollorderseqcolumn"])
+
+    # Open fsm files, where there may be one or two files to open
+    fsm_pattern = re.compile(r"school_census_fsmperiod\.csv$")
+    fsm_onroll_pattern = re.compile(r"school_census_fsmperiodsonroll\.csv$")
+    fsm_patterns = [fsm_pattern, fsm_onroll_pattern]
+    fsm_on = open_and_concat_patterns(files, fsm_patterns, session_folder)
+
+    # Open sen file if exists (only in spring)
+    sen_on = None
+    if "school_census_senneeds.csv" in files:
+        sen_on = open_file(session_folder, "school_census_senneeds.csv")
+    
+    # Create demographics output
+    demographics_on = create_demographics_output(pupil_on, "pupilonrolltableid", addresses_on, fsm_on, sen_on)
+    outputs["onroll_children"] = demographics_on
+
+    # Create sessions outputs
+    pupil_on_termly_sessions = pupil_on[["pupilonrolltableid", "NativeId", "termlysessionspossible", "LA", "Year", "Term", "Acad/LA"]].copy()
+    termly_sessions_on = open_file(session_folder, "school_census_termlysessiondetailsonroll.csv")
+    termly_sessions_on = create_sessions_output(pupil=pupil_on_termly_sessions, identifier="pupilonrolltableid", session_identifier="termlysessionspossible", sessions=termly_sessions_on)
+    outputs["onroll_termly_attendance"] = termly_sessions_on
+
+    if "school_census_summerhalfterm2sessiondetailsonroll.csv" in files:
+        pupil_on_summer_sessions = pupil_on[["pupilonrolltableid", "NativeId", "summerhalfterm2sessionspossible", "LA", "Year", "Term", "Acad/LA"]].copy()
+        summer_sessions_on = open_file(session_folder, "school_census_summerhalfterm2sessiondetailsonroll.csv")
+        summer_sessions_on = create_sessions_output(pupil=pupil_on_summer_sessions, identifier="pupilonrolltableid", session_identifier="summerhalfterm2sessionspossible", sessions=summer_sessions_on)
+        outputs["onroll_summer_attendance"] = summer_sessions_on
+
+    # Create exclusions output
+    exclusions_on = open_file(session_folder, "school_census_termlyexclusionsonroll.csv")
+    outputs["onroll_exclusions"] = exclusions_on
+
+    # Off roll tables
+    # Make demographics output
+    pupil_off = open_file(session_folder, "school_census_pupilnolongeronroll.csv")
+    addresses_off = open_file(session_folder, "school_census_addressesoffroll.csv")
```

Comment:
```suggestion
    if "school_census_addressesoffroll.csv" in files:
        addresses_off = open_file(session_folder, "school_census_addressesoffroll.csv")
    else:
        addresses_off = None
```
deals with addressesoffroll not being available (alongside `create_demographics_output` change also suggested)